### DISCUSSION TITLE
Coincidenceally named a seb file SEBClientSettingsSomething.seb - now SEB always starts with those settings

### DISCUSSION THREAD

**Discussion by raphael303** [SafeExamBrowser/seb-mac#282] [General]
I named a seb file for an exam: SEBClientSettingsSomething.seb. This particular file doesn't allow accessing the SEBConfiguration.

Since I had opened it, every time I open SEB it starts with this file. And I can not go into settings to do anything.
If I open another seb file which previously allowed accessing the settings, I can also not access the settings (greyed out).

I have renamed the original file (SEBClientSettingsSomething to SEBSettingsSomething). No change. It starts with this file, even if I click the SEB-Icon in the taskbar
I have checked /library/preferences (user and system library) for the SEBClientSettings.seb. There is none.
I have deleted org.safeexambrowser.SafeExamBrowser folder in the library, delete SEB, reinstalled SEB. No change.
I have taken one of my older seb files, which allow for settings-access and named them SEBClientSettings.seb and placed it into library/preferences. SEB crashes on startup and automatically deletes the file. The file works fine if I open it directly. just without Settings access.

So, I'm stuck now. I can't use SEB for anything, I can't reinstall it, can't edit any older .seb files, can't create any new ones.

Any help is appreciated.
OSX Ventura, 13.1, SEB downloaded today.

**Comment by danschlet** [comment]
I don't understand what exactly caused your issue. But if you read the [manual](https://safeexambrowser.org/macosx/mac_usermanual_en.html#ConfigDeployment) or [FAQ](https://safeexambrowser.org/faq/faq_en.html#MacOSX), you would find out how to reset SEB settings:

5. Open the Terminal app (you can use the Spotlight search icon on the upper right in your menu bar to find Terminal).
6. Type/copy paste this text: `defaults delete org.safeexambrowser.SafeExamBrowser` (for SEB 2.3 and newer) or `defaults delete org.safeexambrowser.Safe-Exam-Browser` (SEB versions before SEB 2.3) This deletes the client configuration of the Safe Exam Browser application, so when starting it is showing the same information page like when it was installed first

**Reply by raphael303** [reply]
Thank you! this worked.
imho, this is sort of a bug, even if probably not going to be a problem often. Also, on the same system: After using SEB Safari does not connect to the internet anymore (chrome does). Safari has to be restarted. This is every time.

**Reply by danschlet** [reply]
No, this is no bug. There are two types of config files, one configures the client permanently. You just used a wrong configuration I guess. Again, read the manual (section Configuration). There it is explained.

Also Safari with default settings doesn’t need to be restarted, as long as SEB is quit normally and doesn’t crash.

**Comment by raphael303** [comment]
Well, I disagree. I just made a config-file, saved it in a folder in my desktop and named it "SEBClientConfigurationMySchool.seb".
After that, SEB kept opening this file, even after I renamed and deleted it. And it applied its setting to any other file I opened.
I don't see how this makes any sense as a feature.
Sure, in my case it was doubly annoying, because the file had config-access disabled. But it doesn't change the fact, that SEB took the file and used it as default setting, without asking and only because the first part of the file name is the same as the config file in preferences.
Also, the fact, that it didn't accept another, new config file in the library/preferences folder and didn't create a new one in there, when there was none, doesn't seem to be alright.

**Reply by danschlet** [reply]
**This is NOT possible with a standard SEB**. SEB never "just keeps opening" a config file, in fact it never on its own opens any config file, unless [as described here](https://safeexambrowser.org/macosx/mac_usermanual_en.html#ConfigDeployment) if it is named **SEBClientSettings.seb** (exactly like this, not with any suffixes) and placed 

- the Preferences folder of the machine (/Library/Preferences/), or if none is found there, then
- the Preferences folder of the current user (~/Library/Preferences/).

So either you're using a non-official SEB version with some weird added feature or some third party tool which for example tries to open any file on the desktop with the according application (wouldn't know if something like that exists) or some Shortcut with this functionality maybe?

You can make a video if you want to proove that really happens for you. But sorry, I wrote every single line of SEB for macOS/iOS core code, so trust me, I would know if there is such a feature you're describing.

**Reply by raphael303** [reply]
Well, good for you, you are so sure about yourself. I have written 1000s of line of code myself and the one thing I know for sure is, that there are ALWAYS things happening I did't think they could.

Anyhow. My SEB, as mentioned above, was downloaded today from the official SEB website. The problem persisted. The version I used before was also downloaded from the SEB website, just a couple of months ago.

Yes, I read exactly the part of the manual you posted above. And it's why I went to check the preferences folders. There was nothing in it. As it should be. Nevertheless SEB AT EVERY start with the Program Icon or even with double click on another .seb file opened with the settings of the file named SEBClientSettingsMyschool.seb. Up to the terminal command absolutely nothing I did stopped SEB from doing this.

Look. this is just what happened to me. You can deny it happening or say it is not possible, but it will not change the fact, that it happened here exactly as I described.

Feel free to test it yourself with a file named as mine on a ventura mac. Save it in icloud, which is in fact a folder somewhere within the library. Open it with a double click. close it. open SEB with the icon. Check if it happens too.

**Reply by danschlet** [reply]
Well, SEB just never opens a config file when it is started directly, with its icon, that's what I'm sure about. But I realized, what you initially described makes totally sense, and also that it stopped after you used the command to delete those client settings. You just interpreted it differently because you don't know how exactly SEB uses system preferences and config files:

Started with the app icon, SEB opens its client config, which is saved into the NSUserDefaults database, which itself uses the org.safeexambrowser.SafeExamBrowser.plist file in the ~/Library/Preferences/ folder as input (but is cashed in a system database, that's why the `defaults delete ...` command is necessary to get rid of that).

When you saved that file, you used the default setting **Use SEB settings file for ... configuring a client** in the [Config File Pane](https://safeexambrowser.org/macosx/mac_usermanual_en.html#ConfigFilePane) (you can attach the file if you want me to confirm that). If you want to have a config file which connects SEB to an exam, but not permanently, then you have to select **Use SEB settings file for ... starting an exam**.

**Reply by raphael303** [reply]
[SEBSettingsStadtschulenSolothurn.seb.zip](https://github.com/SafeExamBrowser/seb-mac/files/10921877/SEBSettingsStadtschulenSolothurn.seb.zip)
I believe I checked starting an exam. But feel free to check
